types/features/features.go

```diff
@@ -53,13 +53,21 @@ type Seccomp struct {
 	// Nil value means "unknown", not "no support for any action".
 	Actions []string `json:"actions,omitempty"`
 
-	// Operators is the list of the recognized actions, e.g., "SCMP_CMP_NE".
+	// Operators is the list of the recognized operators, e.g., "SCMP_CMP_NE".
 	// Nil value means "unknown", not "no support for any operator".
 	Operators []string `json:"operators,omitempty"`
 
-	// Operators is the list of the recognized archs, e.g., "SCMP_ARCH_X86_64".
+	// Archs is the list of the recognized archs, e.g., "SCMP_ARCH_X86_64".
 	// Nil value means "unknown", not "no support for any arch".
 	Archs []string `json:"archs,omitempty"`
+
+	// KnownFlags is the list of the recognized filter flags, e.g., "SECCOMP_FILTER_FLAG_LOG".
+	// Nil value means "unknown", not "no support for any flag".
+	KnownFlags []string `json:"known_flags,omitempty"`
+
+	// Flags is the list of the supported filter flags, e.g., "SECCOMP_FILTER_FLAG_LOG".
```

Comment:
Doesn't match the variable name

Reply:
How is "supported" different from "recognized"?

Having two concepts doesn't seem consistent with other properties in this JSON

Reply:
oops, fixed

Reply:
In this context "recognized" means known, i.e. runc is aware of the flag (the flag is known to runc).

OTOH "supported" means this flag is not merely known to (recognized by) runc, but is also supported by the currently used libseccomp and the running kernel.

If you don't like such distinction, we can only print one set -- choose which one makes more sense. To be, "supported" makes more sense than "known" as despite the flag being known you can't use it anyway. The downside is you don't know if more flags would be supported after e.g. a kernel upgrade.

Reply:
@AkihiroSuda PTAL 🙏🏻

Reply:
@AkihiroSuda let me add some more context here.

All the "features" that we have in `runc features` are "known features", meaning they are recognized by this runc build. Some of those (like mount flags, for example) can not be used because the kernel version is too old, some library version is old, or a binary is missing. So, the feature is known but not supported.

For the sake of the seccomp flags test case, I need to have seccomp flags that are supported by the current runtime (meaning runc + libraries + kernel), so I added a distinction between "known" and "supported" flags.

I guess that the use case for this (distinguishing between known and supported flags) can also be used by the upper level (container orchestration) to query the runtime platform capabilities. Say, some mount attributes (like "rro") might be known but not supported due to older kernel.

I agree that this distinction doesn't really fit into current output of `runc features`, since we print known (rather than supported) features and flags.

Do you have any idea how to add such a distinction (between known and supported), if we ever want to add it? Something like `runc features --supported` (and maybe `--known` for the sake of completeness?).

It is really hard to write a seccomp flags test (in bash, that is) without relying on `runc features` to print the list of supported seccomp flags, but I guess this is a separate issue. Perhaps I can rewrite the test in Go (and make it the first runc integration test written in Go).